assets/js/libs/experiment.js

```diff
@@ -0,0 +1,51 @@
+(function(){
+  window.Experiment = function(experimentName, experimentClass) {
+    this.EXPERIMENT_NAME = experimentName;
+    this.variants = [];
+    this.experimentClass = experimentClass;
+
+    this.addVariant = function(name, weight) {
+      if(weight < 1) { return alert('Snowplow requires weights to be integers.'); }
+      this.variants.push({
+        experiment_name: this.EXPERIMENT_NAME,
+        name: name,
+        weight: weight
+      })
+    }
+
+    this.chooseVariant = function() {
+      var savedVariant = window.localStorage.getItem(this.EXPERIMENT_NAME);
+      if(savedVariant) { return savedVariant }
+
+      // Use a CDF to select a variant based on weight.
+      var total = this.variants.reduce(function(total, variant) {
+        return total + variant.weight;
+      }, 0);
+      var random = Math.floor(Math.random() * (total + 1));
+      var selectedVariant = this.variants.find(function(variant) {
+        random -= variant.weight;
```

Comment:
To me it's a bit weird to mutate a value like this, but I guess it works.

A different way to tackle this, is to create a list of the cumulative weights (e.g. `const sums = [50, 100]` and use `sums.findIndex(x => x < Math.random() * sums[sums.length - 1])` or something.

I think either is fine though, just a remark (you usually don't expect the callback in `find` to mutate a value.